common4j/src/main/com/microsoft/identity/common/java/opentelemetry/TextMapPropagatorExtension.java

```diff
@@ -0,0 +1,101 @@
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+package com.microsoft.identity.common.java.opentelemetry;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import io.opentelemetry.api.baggage.propagation.W3CBaggagePropagator;
+import io.opentelemetry.api.trace.propagation.W3CTraceContextPropagator;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.context.propagation.TextMapGetter;
+import io.opentelemetry.context.propagation.TextMapPropagator;
+import io.opentelemetry.context.propagation.TextMapSetter;
+
+/**
+ * Extension class for handling OpenTelemetry context propagation.
+ * Provides utility methods for injecting and extracting context which contains Baggage and SpanContext.
+ */
+public final class TextMapPropagatorExtension {
+
+    private static final TextMapPropagator PROPAGATOR = TextMapPropagator.composite(
```

Comment:
Can you check this (and any new otel code introduced in this PR i.e. methods we were previously not using) if they need to wrapped in a try/catch block to make it safe for older devices. This is what the other "extension" classes written for otel are doing to protect against exceptions that fall into the MethodNotAvailable sort of category on older devices.

Reply:
Noted; I added try/catch blocks in the methods of the new extension classes I added, as well as new methods I added to existing extension classes.